workshop_R.Rmd

```diff
@@ -0,0 +1,779 @@
+```
+student_results_aggregated_suppressed_EES <- student_results_aggregated_suppressed %>%
+  mutate(time_identifier = 'Academic year', geographic_level = 'Regional',
+         region_name = case_when(school == 'MS' ~ "East of England",
+                                 school == 'GP' ~ 'London')) %>%
+  rename(time_period = year)
+  
+```
+The above code gets us the correct geographic level and region name, but not region code. We can use a *look-up table* to get the correct code...
+
+## Joins & joining geographic codes via lookup tables
+
+The [dfe-analytical-services](https://github.com/dfe-analytical-services) area on GitHub holds all publicly available code from DfE. The code behind the [EES data screener](https://rsconnect/rsc/dfe-published-data-qa/) is available here, and [the screener repository](https://github.com/dfe-analytical-services/dfe-published-data-qa) contains a [folder full of geographic lookup tables](https://github.com/dfe-analytical-services/dfe-published-data-qa/tree/master/data)! 
+
+Today we will use the [regions.csv](https://github.com/dfe-analytical-services/dfe-published-data-qa/blob/master/data/regions.csv) file from this folder. We can pull the csv directly from the repo using the 'raw' URL: https://raw.githubusercontent.com/dfe-analytical-services/dfe-published-data-qa/master/data/regions.csv. 
+Using this also means if the CSV is updated on GitHub, when you run your code it will automatically pull the latest version of the CSV! Adding appropriate headings and comments, pull in this lookup table using the following code:
+
+```
+regions_lookup <- read.csv("https://raw.githubusercontent.com/dfe-analytical-services/dfe-published-data-qa/master/data/regions.csv")
+```
+
+The lookup table should have appeared in the environment! Click on the name to open it, or use the code `View(regions_lookup)`. Now we can join the codes to our data using the region name using the `left_join()` function! We can 'pipe' joins in the exact same way we have piped summarise, group_by, mutate and rename statements. In the console, type `?join` to see the different types of join within the `dplyr` package!
+
+Here we will use a left join because we want to keep all of the rows from our data, and only join on rows that match from the lookup table. We can use the following code to join the region code using region name and reorder the columns:
+
+```
+student_results_aggregated_suppressed_EES <- student_results_aggregated_suppressed %>%
+  mutate(time_identifier = 'Academic year', geographic_level = 'Regional',
+         region_name = case_when(school == 'MS' ~ "East of England",
+                                 school == 'GP' ~ 'London')) %>%
+  rename(time_period = year) %>%
+  left_join(regions_lookup, by = c('region_name' = 'region_name')) %>%
+  select(time_period, time_identifier, geographic_level, region_name, region_code, sex, age, students, g1_mean, g2_mean, g3_mean)
```

Comment:
This select() function also getting cut off when I try and copy and paste. Worth splitting over a few lines:
select(
time_period, time_identifier, 
geographic_level, region_name, region_code, 
sex, age, 
students, g1_mean, g2_mean, g3_mean
)